### ISSUE TITLE
Cyrillic UTF-8 columns & data fails with "column name cannot be empty"

### ISSUE THREAD

**Issue by ilvez** [infused/dbf#103]
I have an issue with specific DBF file: [issue.zip](https://github.com/infused/dbf/files/14946676/issue.zip) (zipped it to attach to ticket).

This file is constructed and is not the original file, which I can't share. But it crashes similarly and I used same characters from original file (cyrrillic columns & data).

My repro script is very simple:
```ruby
require 'dbf'

DBF::Table.new(File.open('issue.dbf')).each { |w| puts w }
```

Tested with latest version & with older versions (until 3.0.0) as well, they all fail with `column name cannot be empty`

Stacktrace:
```
/home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/column.rb:114:in `validate_name': column name cannot be empty (DBF::Column::NameError)
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/column.rb:50:in `initialize'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/table.rb:230:in `new'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/table.rb:230:in `block (2 levels) in build_columns'
	from <internal:kernel>:90:in `tap'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/table.rb:219:in `block in build_columns'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/table.rb:313:in `safe_seek'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/table.rb:217:in `build_columns'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/table.rb:112:in `columns'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/table.rb:185:in `record'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/table.rb:120:in `block in each'
	from <internal:numeric>:237:in `times'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/dbf-4.3.2/lib/dbf/table.rb:120:in `each'
	from (irb):3:in `<main>'
	from <internal:kernel>:187:in `loop'
	from /home/ilvez/.local/share/mise/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/irb-1.12.0/exe/irb:9:in `<top (required)>'
	from /home/ilvez/.local/share/mise/installs/ruby/latest/bin/irb:25:in `load'
```

**Comment by ilvez** [comment]
I can read the same file with using Python `dbfread` library or `shapelib` command line tools.

**Comment by infused** [comment]
Thanks for reporting this bug. Currently, the gem restricts column names to ASCII only. Instead, it should respect the file's embedded encoding or the encoding argument specified when opening it. I've started on a fix, but it may take me a few days to get a new release out the door.

**Comment by ilvez** [comment]
Thank you for quick response & effort to fix it!

**Comment by infused** [comment]
I've released version 5.0.0 that fixes this issue. Please give it a try and confirm that it works for you.

**Comment by ilvez** [comment]
Thank you so much. It's working for me!
